### DISCUSSION TITLE
How stresses/strains are outputted?

### DISCUSSION THREAD

**Discussion by hchen139** [idaholab/moose#26005] [Q&A General]
Hi,
I'm working on postprocessing stresses/strains from MOOSE simulation. For each element, the stresses/strains are calculated at _qp points of an element. However, when output the stress/strain values using Auxkernel, a single value is outputted for an element. I wonder how this single value for an element is determined. Is this an "average" value based on the _qp values of an element? If so, what is the averaging algorithm? Also, is it possible to output all the values for all the _qp points of an element?

Thanks for the help.

**Comment by richmondodufisan** [comment]
How are you getting the output from an AuxKernel? To my understanding those just tell MOOSE what to calculate for an AuxVariable over the domain that variable is defined.

On the other hand, a Postprocessor block is what I use to output what I need. If there's a single value being output maybe you can check that block of your input file and see if its a PointValue/NodalSum/ElementAverageValue etc? The full documentation of Postprocessor objects is here: https://mooseframework.inl.gov/syntax/Postprocessors/

**Comment by GiudGiud** [comment]
Hello

So if using an AuxKernel, the output is based on the finite element family used for the auxiliary variables. 
The value of the property is computed at all Qps then there is a least-square solve to pick the coefficients multiplying the shape functions for the variable. So essentially a projection happens under the hood. 

By default we use constant monomials to output the material properties.  There are material property output parameters on each Material object

Guillaume
